src/ring_theory/etale.lean

```diff
@@ -244,6 +276,67 @@ formally_etale.iff_unramified_and_smooth.mpr
 
 end comp
 
+section of_surjective
+
+variables {R S : Type u} [comm_ring R] [comm_semiring S]
+variables {P A : Type u} [comm_ring A] [algebra R A] [comm_ring P] [algebra R P]
+variables (I : ideal P) (f : P →ₐ[R] A) (hf : function.surjective f)
+
+lemma formally_smooth.of_split [formally_smooth R P] (g : A →ₐ[R] P ⧸ f.to_ring_hom.ker ^ 2)
+  (hg : f.ker_square_lift.comp g = alg_hom.id R A) :
+  formally_smooth R A :=
+begin
+  constructor,
+  introsI C _ _ I hI i,
+  let l : P ⧸ f.to_ring_hom.ker ^ 2 →ₐ[R] C,
+  { refine ideal.quotient.liftₐ _ (formally_smooth.lift I ⟨2, hI⟩ (i.comp f)) _,
+    have : ring_hom.ker f ≤ I.comap (formally_smooth.lift I ⟨2, hI⟩ (i.comp f)),
+    { rintros x (hx : f x = 0),
+      have : _ = i (f x) := (formally_smooth.mk_lift I ⟨2, hI⟩ (i.comp f) x : _),
+      rwa [hx, map_zero, ← ideal.quotient.mk_eq_mk, submodule.quotient.mk_eq_zero] at this },
+    intros x hx,
+    have := (ideal.pow_mono this 2).trans (ideal.le_comap_pow _ 2) hx,
+    rwa hI at this },
+  have : i.comp f.ker_square_lift = (ideal.quotient.mkₐ R _).comp l,
+  { apply alg_hom.coe_ring_hom_injective,
+    apply ideal.quotient.ring_hom_ext,
+    ext x,
+    exact (formally_smooth.mk_lift I ⟨2, hI⟩ (i.comp f) x).symm },
+  exact ⟨l.comp g, by rw [← alg_hom.comp_assoc, ← this, alg_hom.comp_assoc, hg, alg_hom.comp_id]⟩
+end
+
+include hf
+
+/-- Let `P →ₐ[R] A` be a surjection with kernel `J`, and `P` is a formally smooth `R`-algebra,
+then `A` is formally smooth over `R` iff the surjection `P ⧸ J ^ 2 →ₐ[R] A` has a section. -/
```

Comment:
Do you think you could add a sentence (or two) to this docstring about the geometric intuition behind the statement?

Reply:
I don't think I have a good geometric intuition behind this. I'd be more than happy to add such a remark if you could help to provide one though.

Reply:
So `Spec(A)` is a closed subscheme of `Spec(P)`, cut out by `J`. And `Spec(P/J^2)` is an example of a "nilpotent thickening" of `Spec(A)`. Saying that the section exists, means that you can retract the thickening back onto `Spec(A)`.
If `Spec(P)` has crazy singularities, nothing is stopping you from those singularities ending up in `Spec(A)`. But if `Spec(P)` is smooth, then the existence of that retraction means that every "tangent vector" nicely retracts onto `Spec(A)`. Whereas if you something like `J = (xy)`, and you get the union of two coordinate axes, then you can imagine there is no way to algebraically retract a thickening of the coordinate axes back onto the axes. You get in trouble at the origin.

Reply:
```suggestion
then `A` is formally smooth over `R` iff the surjection `P ⧸ J ^ 2 →ₐ[R] A` has a section.

 Geometric intuition: we require that a nilpotent thickening of `Spec A` inside `Spec P` admits a retraction. -/
```